pkg/authorization/apibinding_authorizer.go

```diff
@@ -33,50 +32,25 @@ import (
 
 	apisv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/apis/v1alpha1"
 	kcpinformers "github.com/kcp-dev/kcp/pkg/client/informers/externalversions"
+	"github.com/kcp-dev/kcp/pkg/indexers"
 	rbacwrapper "github.com/kcp-dev/kcp/pkg/virtual/framework/wrappers/rbac"
 )
 
-const (
-	byWorkspaceIndex = "apiBindingAuthorizer-byWorkspace"
-)
-
 const (
 	APIBindingContentAuditPrefix   = "apibinding.authorization.kcp.dev/"
 	APIBindingContentAuditDecision = APIBindingContentAuditPrefix + "decision"
 	APIBindingContentAuditReason   = APIBindingContentAuditPrefix + "reason"
 )
 
-// NewAPIBindingAccessAuthorizer returns an authorizer that checks if the the request is for a
+// NewAPIBindingAccessAuthorizer returns an authorizer that checks if the request is for a
 // bound resource or not. If the resource is bound we will check the user has RBAC access in the
-// exported resources workspace. If it is not allowed we will return NoDecision, if allowed we
+// exported resource workspace. If it is not allowed we will return NoDecision, if allowed we
 // will call the delegate authorizer.
 func NewAPIBindingAccessAuthorizer(kubeInformers kubernetesinformers.SharedInformerFactory, kcpInformers kcpinformers.SharedInformerFactory, delegate authorizer.Authorizer) (authorizer.Authorizer, error) {
-	if _, found := kcpInformers.Apis().V1alpha1().APIBindings().Informer().GetIndexer().GetIndexers()[byWorkspaceIndex]; !found {
-		err := kcpInformers.Apis().V1alpha1().APIBindings().Informer().AddIndexers(
-			cache.Indexers{
-				byWorkspaceIndex: func(obj interface{}) ([]string, error) {
-					return []string{logicalcluster.From(obj.(metav1.Object)).String()}, nil
-				},
-			},
-		)
-		if err != nil {
-			// nothing we can do here. But this should also never happen. We check for existence before.
-			return nil, fmt.Errorf("failed to add indexer for APIBindings: %w", err)
-		}
-	}
-	if _, found := kcpInformers.Apis().V1alpha1().APIExports().Informer().GetIndexer().GetIndexers()[byWorkspaceIndex]; !found {
-		err := kcpInformers.Apis().V1alpha1().APIExports().Informer().AddIndexers(
-			cache.Indexers{
-				byWorkspaceIndex: func(obj interface{}) ([]string, error) {
-					return []string{logicalcluster.From(obj.(metav1.Object)).String()}, nil
-				},
-			},
-		)
-		if err != nil {
-			// nothing we can do here. But this should also never happen. We check for existence before.
-			return nil, fmt.Errorf("failed to add indexer for APIExports: %w", err)
-		}
-	}
+	apiBindingIndexer := kcpInformers.Apis().V1alpha1().APIBindings().Informer().GetIndexer()
+	apiExportIndexer := kcpInformers.Apis().V1alpha1().APIExports().Informer().GetIndexer()
+	indexers.AddIfNotPresentOrDie(apiBindingIndexer, indexers.ClusterScoped())
+	indexers.AddIfNotPresentOrDie(apiExportIndexer, indexers.ClusterScoped())
```

Comment:
I don't think you need these - see https://github.com/kcp-dev/kcp/blob/ea31363ae8d683175e435421c4597646e1a317e8/pkg/server/config.go#L230-L235